php/class-pwawp-app-manifest.php

```diff
@@ -0,0 +1,171 @@
+<?php
+/**
+ * PWAWP_APP_Manifest class.
+ *
+ * @package PWA
+ */
+
+/**
+ * PWAWP_APP_Manifest class.
+ *
+ * Mainly copied from Jetpack_PWA_Manifest and Jetpack_PWA_Helpers.
+ */
+class PWAWP_APP_Manifest {
+
+	/**
+	 * The query arg that is present in a request for the app manifest.
+	 *
+	 * @var string
+	 */
+	const MANIFEST_QUERY_ARG = 'pwawp_manifest';
+
+	/**
+	 * The theme color to use if no dynamic values are present.
+	 *
+	 * @var string
+	 */
+	const FALLBACK_THEME_COLOR = '#fff';
+
+	/**
+	 * The default manifest icon sizes.
+	 *
+	 * Copied from Jetpack_PWA_Helpers::get_default_manifest_icon_sizes().
+	 * Based on a conversation in https://github.com/GoogleChrome/lighthouse/issues/291
+	 *
+	 * @var int[]
+	 */
+	public $default_manifest_icon_sizes = array( 192, 512 );
+
+	/**
+	 * Inits the class.
+	 *
+	 * Mainly copied from Jetpack_PWA_Manifest::__construct().
+	 */
+	public function init() {
+		add_action( 'wp_head', array( $this, 'manifest_link_and_meta' ) );
+		add_action( 'amp_post_template_head', array( $this, 'manifest_link_and_meta' ) );
+		add_action( 'template_redirect', array( $this, 'send_manifest_json' ), 2 );
+	}
+
+	/**
+	 * Outputs the <link> and <meta> tags for the app manifest.
+	 *
+	 * Mainly copied from Jetpack_PWA_Manifest::render_manifest_link().
+	 */
+	public function manifest_link_and_meta() {
+		?>
+		<link rel="manifest" href="<?php echo esc_url( add_query_arg( self::MANIFEST_QUERY_ARG, '1', home_url( '/' ) ) ); ?>">
+		<meta name="theme-color" content="<?php echo esc_attr( $this->get_theme_color() ); ?>">
```

Comment:
Is it better to define this in meta vs the manifest?

Reply:
Good question. I'm not sure, but it looks like it's suggested to have both:
>The audit passes if Lighthouse finds a theme-color meta tag in the page's HTML and a theme_color property in the Web App Manifest. Lighthouse does not test whether the values are valid CSS color values.

([source](https://developers.google.com/web/tools/lighthouse/audits/address-bar#more-info))
 
But this suggests that it ignores the manifest `theme-color` if it's present in the `<meta>` tag:

>If you page already has a theme-color meta tag — for example <meta name="theme-color" content="#2196F3"> — then the page level configuration will be used instead of the value in the manifest.

([source](https://developers.google.com/web/updates/2015/08/using-manifest-to-set-sitewide-theme-color))